htlcswitch/switch.go

```diff
@@ -1287,12 +1348,63 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 // checkCircularForward checks whether a forward is circular (arrives and
 // departs on the same link) and returns a link error if the switch is
 // configured to disallow this behaviour.
-func checkCircularForward(incoming, outgoing lnwire.ShortChannelID,
+func (s *Switch) checkCircularForward(incoming, outgoing lnwire.ShortChannelID,
 	allowCircular bool, paymentHash lntypes.Hash) *LinkError {
 
-	// If the route is not circular we do not need to perform any further
```

Comment:
Hmm, I thought a prior version of this _also_ checked the payment hash as well. IMO that' another relevant thing we can do in order to mitigate loop attacks somewhat. We'll need to take care to not disallow MPP payment splits along the way.

Reply:
It takes the payment hash and just logs it